packages/dds/tree/src/test/shared-tree/sharedTreeFuzzTests.ts

```diff
@@ -0,0 +1,273 @@
+    }
+    return finalState;
+}
+
+function applyFuzzChange(
+    tree: ISharedTree,
+    contents: FuzzChange,
+    transactionResult: TransactionResult,
+): void {
+    switch (contents.fuzzType) {
+        case "insert":
+            tree.runTransaction((forest, editor) => {
+                const field = editor.sequenceField(contents.parent, contents.field);
+                field.insert(
+                    contents.index,
+                    singleTextCursor({ type: brand("Test"), value: contents.value }),
+                );
+                return transactionResult;
+            });
+            break;
+        case "delete":
+            // generated edit returns path as undefined if no node exists
+            if (contents.path !== undefined) {
+                const parent = contents.path?.parent;
+                const delField = contents.path?.parentField;
+                const parentIndex = contents.path?.parentIndex;
+                tree.runTransaction((forest, editor) => {
+                    const field = editor.sequenceField(parent, delField);
+                    field.delete(parentIndex, 1);
+                    return transactionResult;
+                });
+            }
+            break;
+        case "setPayload":
+            // generated edit returns path as undefined if no node exists
+            if (contents.path !== undefined) {
+                const path = contents.path;
+                tree.runTransaction((forest, editor) => {
+                    editor.setValue(path, contents.value);
+                    return transactionResult;
+                });
+            }
+            break;
+        default:
+            fail("Invalid edit.");
+    }
+}
+
+function runBatch(
+    opGenerator: () => AsyncGenerator<Operation, FuzzTestState>,
+    fuzzActions: (
+        generatorFactory: AsyncGenerator<Operation, FuzzTestState>,
+        seed: number,
+    ) => Promise<FuzzTestState>,
+    opsPerRun: number,
+    batchSize: number,
+    random: IRandom,
+    skipTests?: boolean,
+): void {
+    const seed = random.integer(1, 1000000);
+    for (let i = 0; i < batchSize; i++) {
+        const runSeed = seed + i;
+        const generatorFactory = () => take(opsPerRun, opGenerator());
+        if (skipTests) {
+            it.skip(`with seed ${runSeed}`, async () => {
+                await fuzzActions(generatorFactory(), runSeed);
+            }).timeout(20000);
+        } else {
+            it(`with seed ${runSeed}`, async () => {
+                await fuzzActions(generatorFactory(), runSeed);
+            }).timeout(20000);
+        }
+    }
+}
+
+export function runSharedTreeFuzzTests(title: string): void {
+    const random = makeRandom(0);
+    const testBatchSize = 20;
+    describeFuzz(title, () => {
+        const testOpsPerRun = 20;
+        describe("basic convergence", () => {
+            describe("using TestTreeProvider", () => {
+                describe(`with stepSize ${testOpsPerRun}`, () => {
```

Comment:
I don't think these two inner levels of `describe` add much value. If we add more tests in the future and we need to differentiate them the we'll reconsider the question then.
(Same for the describe calls under "abort all edits" below)